s3tests/common.py

```diff
@@ -53,6 +53,29 @@ def nuke_bucket(bucket):
                 key.set_canned_acl('private')
                 key.delete()
                 deleted_cnt += 1
+        # Delete old versioned keys
+        deleted_cnt = 1
+        while deleted_cnt:
+            deleted_cnt = 0
+            for key in bucket.list_versions():
+                print 'Cleaning bucket {bucket} key {key} version {verion}'.format(
+                    bucket=bucket,
+                    key=key,
+                    version=key.version_id,
+                    )
+                key.set_canned_acl('private')
```

Comment:
- the ACL setting is copied from the original block, and seems to be rooted in fixing cases of weird bucket/key ownership from the test suite (the bucket owner gets permission back to the object as a result of changing the ACL).
- If the S3 implementation does not support versioned objects, then `list_versions` is not usable.